crates/egui-wgpu/src/egui.wgsl

```diff
@@ -95,10 +98,35 @@ fn vs_main(
 @group(1) @binding(0) var r_tex_color: texture_2d<f32>;
 @group(1) @binding(1) var r_tex_sampler: sampler;
 
+fn sample_texture(in: VertexOutput) -> vec4<f32> {
+    if r_locals.predictable_texture_filtering == 0 {
+        // Hardware filtering: fast, but varies across GPUs and drivers.
+        return textureSample(r_tex_color, r_tex_sampler, in.tex_coord);
+    } else {
+        // Manual bilinear filtering with four taps at pixel centers
+        let texture_size = vec2<f32>(textureDimensions(r_tex_color, 0));
+        let pixel_coord = in.tex_coord * texture_size - 0.5;
+        let pixel_fract = fract(pixel_coord);
+        let pixel_floor = floor(pixel_coord);
+
+        // Sample at four pixel centers
+        let texel_size = 1.0 / texture_size;
+        let tl = textureSample(r_tex_color, r_tex_sampler, (pixel_floor + vec2<f32>(0.5, 0.5)) * texel_size);
+        let tr = textureSample(r_tex_color, r_tex_sampler, (pixel_floor + vec2<f32>(1.5, 0.5)) * texel_size);
+        let bl = textureSample(r_tex_color, r_tex_sampler, (pixel_floor + vec2<f32>(0.5, 1.5)) * texel_size);
+        let br = textureSample(r_tex_color, r_tex_sampler, (pixel_floor + vec2<f32>(1.5, 1.5)) * texel_size);
```

Comment:
I think `textureSample` still performs a four-tap sampling, or eight-tap if you have miplevels. 
[`textureLoad`](https://www.w3.org/TR/WGSL/#textureload) is the one without filtering.

Reply:
entirely depends on the set sampler and if you hit the pixel exactly in the middle as done here it _should_ not do any filtering, but I agree that `textureLoad`would be the better bet as you're already working in whole-pixel coordinates anyways

Reply:
but then I need to do manual clamping… but sure. Seems safer.